nextcloud-startup-script.sh

```diff
@@ -330,27 +330,58 @@ bash $SCRIPTS/nextcloud_configuration.sh
 bash $SCRIPTS/additional_apps.sh
 
 clear
-# Change passwords
+### Change passwords
 # CLI USER
-print_text_in_color "$ICyan" "For better security, change the system user password for [$(getent group sudo | cut -d: -f4 | cut -d, -f1)]"
-any_key "Press any key to change password for system user..."
+msg_box "For better security, we will now change the password for the UNIX-user for Ubuntu"
+UNIXUSER="$(getent group sudo | cut -d: -f4 | cut -d, -f1)"
 while :
 do
-    sudo passwd "$(getent group sudo | cut -d: -f4 | cut -d, -f1)" && break
+    UNIX_PASSWORD=$(input_box_flow "Please type in the new password for the UNIX-user: [$UNIXUSER]")
+    if [[ "$UNIX_PASSWORD" == *" "* ]]
+    then
+        msg_box "Please don't use spaces"
+    else
+        break
+    fi
 done
-echo
-clear
+check_command echo "$UNIXUSER:$UNIX_PASSWORD" | sudo chpasswd
+unset UNIX_PASSWORD
+
 # NEXTCLOUD USER
 NCADMIN=$(occ_command user:list | awk '{print $3}')
-print_text_in_color "$ICyan" "The current admin user in Nextcloud GUI is [$NCADMIN]"
-print_text_in_color "$ICyan" "We will now replace this user with your own."
-any_key "Press any key to replace the current (local) admin user for Nextcloud..."
-# Create new user
+msg_box "We will now change the username and password for the Nextcloud admin user"
 while :
 do
-    print_text_in_color "$ICyan" "Please enter the username for your new user:"
-    read -r NEWUSER
-    sudo -u www-data $NCPATH/occ user:add "$NEWUSER" -g admin && break
+    NEWUSER=$(input_box_flow "Please type in the name of the Nextcloud admin user. It must differ from [$NCADMIN].\nThe only allowed character are: 'a-z', 'A-Z', '0-9', and '_.@-'")
+    if [[ "$NEWUSER" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
+    elif [ "$NEWUSER" = "$NCADMIN" ]
+    then
+        msg_box "This username is already in use. Please choose a different one."
+    elif [[ "$NEWUSER" =~ [^a-zA-Z0-9_.@-] ]]
+    then
+        msg_box "Allowed characters are: a-z', 'A-Z', '0-9', and '_.@-'\nPlease try again."
+    else
+        break
+    fi
+done
+while :
+do
+    OC_PASS=$(input_box_flow "Please type in the new password for the new nextcloud-admin user $NEWUSER")
+    if [[ "$OC_PASS" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
+    fi
```

Comment:
Here is a mistake: we want to bring up the input-box directly after this was positive and not continue, no?

Reply:
I don't know, you wrote it. :)

I have some time today (the whole day) for testing, and I was planning to work on the Wordpress VM, but I can chime in here as well from time to time.

Reply:
I will check this and merge this today :+1:

Reply:
This prompt is needed as it's for the password, not for the user.

I don't know what yo mean here?